tests/functional/test_rebooting_reboots/__init__.py

```diff
@@ -0,0 +1,69 @@
+from nose import tools
+from nixops.ssh_util import SSHCommandFailed
+from nixops.util import root_dir
+from itertools import product
+from parameterized import parameterized
+
+from tests.functional.shared.deployment_run_command import deployment_run_command
+from tests.functional.shared.create_deployment import create_deployment
+from tests.functional.shared.using_unique_state_file import using_unique_state_file
+
+@parameterized(product(
```

Comment:
You had this kind of expression also in at least one other file. Why do you repeat what seems to be the same code?

Reply:
> You had this kind of expression also in at least one other file

it configuration for test. As you see I have changed the architecture of tests.
Before: tests, that should run on all backends are inherited from https://github.com/NixOS/nixops/blob/aa3602d03cd03fc748cdcf1fc25d97ef863474f0/tests/functional/single_machine_test.py
After: `product` method makes matrix of configurations for tests, `parameterized` (method factory) creates tests

> Why do you repeat what seems to be the same code?

I don't want to extract configuration to one place though it's duplicated because configuration code is susceptible to change